### DISCUSSION TITLE
Zigbee2mqtt container does not show usb passthrough

### DISCUSSION THREAD

**Discussion by Tokn59** [tteck/Proxmox#1388] [Help Needed]
### A clear and concise description of the issue.

In pve the command lsusb reveals :

Bus 001 Device 013: ID 1a86:55d4 QinHeng Electronics SONOFF Zigbee 3.0 USB Dongle Plus V2

In the container (Zigbee2mqtt - debian 11) the command ls -l /dev/serial/by-id : 

![image](https://user-images.githubusercontent.com/9215231/235981570-90f3c136-78c9-4cf6-af86-fe514069a5d7.png)

Container is privileged. Already did reboot etc ..

Any idea whats's wrong ? Thank You !!!!

**Comment by tteck** [comment]
I do not believe it is an issue with the script. I performed a clean installation, and everything appears to be functioning correctly.

```
 _____   _       __             ___   __  _______  ____________
/__  /  (_)___ _/ /_  ___  ___ |__ \ /  |/  / __ \/_  __/_  __/
  / /  / / __  / __ \/ _ \/ _ \__/ // /|_/ / / / / / /   / /   
 / /__/ / /_/ / /_/ /  __/  __/ __// /  / / /_/ / / /   / /    
/____/_/\__, /_.___/\___/\___/____/_/  /_/\___\_\/_/   /_/     
       /____/ 🐝
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 11
Using Container Type: 0
Using Root Password: 05-03-2023
Container ID: 112
Using Hostname: zigbee2mqtt
Using Disk Size: 4
Allocated Cores: 2
Allocated RAM: 1024
Using Bridge: vmbr0
Using IP Address: 192.168.86.252/24
Using Gateway IP Address: 192.168.86.1
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Zigbee2MQTT LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.252 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.113.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Set up Node.js Repository
 ✓ Installed Node.js
 ✓ Set up Zigbee2MQTT Repository
Switch to Edge/dev branch? (y/N) 
 ✓ Installed Zigbee2MQTT
 ✓ Created Service
 ✓ Cleaned
 ✓ Completed Successfully!
```
Proxmox Shell
```
root@server:~# ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 May  1 16:19 usb-Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_a4766cbfb593eb11b7611b4f3d9xxxxx-if00-port0 -> ../../ttyUSB0
root@server:~# 
```
Zigbee2mqtt LXC Console
```
root@zigbee2mqtt:~# ls -l /dev/serial/by-id                                        
total 0
lrwxrwxrwx 1 root root 13 May  1 16:19 usb-Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_a4766cbfb593eb11b7611b4f3d9xxxxx-if00-port0 -> ../../ttyUSB0
root@zigbee2mqtt:~# 
```

**Comment by tteck** [comment]
Run this command `dpkg -s usbutils` in the Proxmox shell and share the output.

**Comment by Tokn59** [comment]
root@zigbee2mqtt:~# dpkg -s usbutils
dpkg-query: package 'usbutils' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.

**Not installed so I installed them ...**

root@zigbee2mqtt:~# apt install usbutils
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libusb-1.0-0
The following NEW packages will be installed:
  libusb-1.0-0 usbutils
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 139 kB of archives.
After this operation, 486 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian bullseye/main amd64 libusb-1.0-0 amd64 2:1.0.24-3 [60.2 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 usbutils amd64 1:013-3 [79.3 kB]
Fetched 139 kB in 0s (679 kB/s)
Selecting previously unselected package libusb-1.0-0:amd64.
(Reading database ... 32558 files and directories currently installed.)
Preparing to unpack .../libusb-1.0-0_2%3a1.0.24-3_amd64.deb ...
Unpacking libusb-1.0-0:amd64 (2:1.0.24-3) ...
Selecting previously unselected package usbutils.
Preparing to unpack .../usbutils_1%3a013-3_amd64.deb ...
Unpacking usbutils (1:013-3) ...
Setting up libusb-1.0-0:amd64 (2:1.0.24-3) ...
Setting up usbutils (1:013-3) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for libc-bin (2.31-13+deb11u6) ...
root@zigbee2mqtt:~# 

**Then ran lsusb:**

root@zigbee2mqtt:~# lsusb
Bus 001 Device 005: ID 04e8:4001 Samsung Electronics Co., Ltd PSSD T7
Bus 001 Device 006: ID 18d1:9302 Google Inc. 
Bus 001 Device 013: ID 1a86:55d4 QinHeng Electronics SONOFF Zigbee 3.0 USB Dongle Plus V2
Bus 001 Device 012: ID 1cf1:0030 Dresden Elektronik ZigBee gateway [ConBee II]
Bus 001 Device 002: ID 8087:8001 Intel Corp. Integrated Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
root@zigbee2mqtt:~#

**Comment by tteck** [comment]
I wanted the command run in the Proxmox shell.

**Comment by tteck** [comment]
What does `ls -l /dev/serial/by-id` show?

**Comment by Tokn59** [comment]
Ok sorry ... here you go ...

root@nucpve:~# dpkg -s usbutils
Package: usbutils
Status: install ok installed
Priority: optional
Section: utils
Installed-Size: 325
Maintainer: Aurelien Jarno <aurel32@debian.org>
Architecture: amd64
Multi-Arch: foreign
Version: 1:013-3
Depends: libc6 (>= 2.14), libudev1 (>= 196), libusb-1.0-0 (>= 2:1.0.22)
Breaks: hwdata (<< 0.334-1~), isenkram (<< 0.45~), kinfocenter (<< 4:5.14.5-2~), libosinfo-1.0-0 (<< 1.8.0-1~), usbip (<< 2.0+5.10.4-1~)
Description: Linux USB utilities
 This package contains the lsusb utility for inspecting the devices
 connected to the USB bus. It shows a graphical representation of the
 devices that are currently plugged in, showing the topology of the
 USB bus. It also displays information on each individual device on
 the bus.
Homepage: https://github.com/gregkh/usbutils
root@nucpve:~#

**Comment by Tokn59** [comment]
sorry about the strikethrough ????

**Comment by Tokn59** [comment]
In proxmox pve: 

root@nucpve:~# ls -l /dev/serial/by-id
ls: cannot access '/dev/serial/by-id': No such file or directory
root@nucpve:~# 

In the lxc :
root@zigbee2mqtt:~# ls -l /dev/serial/by-id
total 0
root@zigbee2mqtt:~#

**Comment by tteck** [comment]
This directory is typically created when you have USB serial devices connected to your system. If the USB serial device is recognized using the command `lsusb` but the `/dev/serial/by-id` directory is still missing, you can try creating the directory manually by running the following command `mkdir -p /dev/serial/by-id`

**Comment by Tokn59** [comment]
Ok, did that and rebooted the lxc. Result: 
root@zigbee2mqtt:~# ls -l /dev/serial/by-id
total 0
root@zigbee2mqtt:~# 

THANK YOU for helping me sort this out. I saw I am not the only one with this issue btw ...

**Comment by tteck** [comment]
Did you run `mkdir -p /dev/serial/by-id` in the Proxmox Shell or LXC console?

**Comment by Tokn59** [comment]
Lxc console …


On 3 May 2023, at 19:31, tteckster ***@***.***> wrote:

﻿

Did you run mkdir -p /dev/serial/by-id in the Proxmox Shell or LXC console?

**Comment by tteck** [comment]
Run it in the Proxmox Shell

**Comment by Tokn59** [comment]
Just did ... and rebooted the lxc as well. Same result though ...

root@zigbee2mqtt:~# ls -l /dev/serial/by-id
total 0
root@zigbee2mqtt:~#

**Comment by tteck** [comment]
Unplug and plug back in the USB stick, and run ` ls -l /dev/serial/by-id` in the **Proxmox Shell**

**Comment by Tokn59** [comment]
Done !

![image](https://user-images.githubusercontent.com/9215231/235999875-1ad9f737-4427-422b-9099-996798fa5b96.png)

**Comment by tteck** [comment]
I'm not certain why your system is not recognizing the device.

**Comment by Tokn59** [comment]
That makes 2 of us .... I am also running HASS OS as a VM and that does use USB Passthrough without any issues. I will continue to search for a cause/solution. Meanwhile .. BIG THANKS for all your help !

Will post if I find solution for sure !

**Comment by tteck** [comment]
Maybe, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1035094 ?

**Comment by tteck** [comment]
https://github.com/Koenkk/zigbee2mqtt/issues/17535#issuecomment-1532241700

**Comment by Tokn59** [comment]
![image](https://user-images.githubusercontent.com/9215231/236027011-7a1ae9d1-ee3a-4a3f-b357-4affc4aa2215.png)

YES !!!!

This did the trick :

 Create `/etc/apt/sources.list.d/bullseye-backports.list` containing
```
# bullseye-backports
deb http://deb.debian.org/debian bullseye-backports main
```
Run `sudo apt update`
Run `sudo apt-get -t bullseye-backports install systemd udev`
reboot

Thanks man !!!!!

**Comment by tteck** [comment]
Did you use backports on Proxmox or the Zigbee2mqtt LXC?

**Comment by Tokn59** [comment]
I wasn't sure so did it on both pve and lxc. Now one step closer but still some issues: 

![image](https://user-images.githubusercontent.com/9215231/236031473-6582fd91-07eb-4bc2-89d3-2064ac7f7a18.png)
****

**Comment by tteck** [comment]
Make sure HAOS is not still using the device.

**Comment by Tokn59** [comment]
It is not in use by HASSOS (Using a Conbee II)
![image](https://user-images.githubusercontent.com/9215231/236034527-6bb56b04-d486-4035-bf2f-c6529e4246ff.png)



Seems user root does not have write access permissions to the serial port. Tried: 
test -w /dev/ttyACM1 && echo success || echo failure. 
Resulted in failure.

**Comment by tteck** [comment]
``` 
  serial:
    port: /dev/serial/by-id/<your device>
    #adapter: deconz	#(uncomment for ConBee II)
    #adapter: ezsp	#(uncomment for ITead Sonoff Zigbee 3.0 USB Dongle Plus V2 model "ZBDongle-E version")
```

**Comment by Tokn59** [comment]
This is my current config : 

![image](https://user-images.githubusercontent.com/9215231/236038171-272647fc-17ce-4689-920a-5506e59791ee.png)

But still :

![image](https://user-images.githubusercontent.com/9215231/236038531-a4d4ba04-cc0a-4463-981d-9b5eb12c1466.png)

**Comment by tteck** [comment]
Your `configuration.yaml` is messed up. Look at the example:
```
frontend:
  port: 9442
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.86.224:1883'
  user: usr
  password: pwd
  keepalive: 60
  reject_unauthorized: true
  version: 4
serial:
  port: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
  #adapter: deconz	#(uncomment for ConBee II)
  #adapter: ezsp        #(uncomment for ITead Sonoff Zigbee 3.0 USB Dongle Plus V2 model "ZBDongle-E version")
advanced:
  pan_id: GENERATE
  network_key: GENERATE
  channel: 20
```

**Comment by Tokn59** [comment]
Corrected (pasted your file and changed with my data). Still get the same error. Checked the Zigbee2mqtt documentation for the specific error "Error: Resource temporarily unavailable Cannot lock port
This error occurs when another program is already using (and thus locking) the adapter. You can find out which via the following command: ls -l /proc/[0-9]/fd/ |grep /dev/ttyACM0 (replace /dev/ttyACM0 with your adapter port)."

So did : ls -l /proc/[0-9]/fd/ |grep /dev/ttyACM1

No processes or programs shown though ...

![image](https://user-images.githubusercontent.com/9215231/236041082-a859df21-fc07-4983-b068-413500dd024d.png)

**Comment by tteck** [comment]
did you uncomment ` #adapter: ezsp`?

**Comment by Tokn59** [comment]
Also did a reboot to make sure nothing is occupying the port ...

![image](https://user-images.githubusercontent.com/9215231/236041562-22e3d17b-8076-48f6-971a-a813c0821fd2.png)

**Comment by tteck** [comment]
Have you used this device before?

**Reply by alamoudimoh** [reply]
What would it be if it is a dongle that i used previously in HA

**Comment by Tokn59** [comment]
No it's brand new. AFAIK it is not the ezsp model but the Sonoff Zigbee 3.0 USB Dongle Plus

My HASS is using a Conbee II but my goal was to move Zigbee2mqtt out of HASS and use the better Sonoff stick.
I will put it ( and YOU) to rest for today. Will try again tomorrow with the Conbvee II stick first.

**Comment by tteck** [comment]
Check for a `cordinator_backup.json` in `/opt/zigbee2mqtt/data/` if found, remove it.

It states `ITead Sonoff Zigbee 3.0 USB Dongle Plus V2` in your devices description. The `-P` (CC2652P chip) model has a 2.5" case, the `-E` (EFR chip) model is much shorter.

https://www.youtube.com/watch?v=iW6D0G_rlSA

**Reply by Tokn59** [reply]
That's correct. Mine is a ZBdongle-E

**Comment by Tokn59** [comment]
Just wanted to let you know that there's nothing wrong with your script. The issue lies with the FW of the dongle in combination with the version of Zigbee2mqtt apparently and/or the passthrough of the USB port(s) to the LXC. I have upgraded the firmware of the dongle (ncp-uart-sw_6.10.3_115200.gbl) and created a new wm with the latest build of HASS OS and the latest version of Zigbee2mqtt ( 1.30.4-1) and here it works perfectly. So ... what is still the issue with the lxc created by the script ? Again thanks for all your support !

**Reply by Tokn59** [reply]
I see that the version in the lxc is still at zigbee2mqtt@1.30.4 What is the best way to update it ? I re-ran the script but that doesn't seem to update it to 1.30.4-1 ...

**Reply by tteck** [reply]
`1.30.4-1` is not a Zigbee2MQTT version, https://github.com/Koenkk/zigbee2mqtt/releases
Must be a Home Assistant thing?

**Comment by Tokn59** [comment]
Ok so now you have 3 options ....

1. Laugh very hard at me
2. Kill me
3. Understand me

First of all there was a real issue with the USB pass through. Solved by what I described above. Secondly last status was I always got this error "Connection not initialized" right ? Here we go .... (RED CHEEKS)

THAT WAS BECAUSE ZIGBEE2MQTT WAS ALREADY RUNNING SUCCESSFULLY !!!!!!! I never though of checking via the URL as I saw this error and assumed (oh dear) it was failing to start !
SHAME ON ME ! SHAME ON ME !!!!

Could you possibly forgive me ?

**Reply by atechnical** [reply]
Don't feel bad ,I just realized after reading your post I DID THE EXACT SAME THING! The directions are a little misleading. THANK YOU for this. I've spent at least 12 hours over the course of a week trying to figure this out.

**Reply by hellms** [reply]
i´m next who spent hours in searching how to fix an issue that that sit in front of the keyboard ;-)
thanks for your heads up post

**Comment by CTMarsh** [comment]
@Tokn59 @tteck Hi both,

Sorry to re-open a closed discussion, but im getting a similar issue.  Related to exactly the same dongle (ZBDongle-E).  I was having the issue that it could not lock the port.  However this has been fixed, but instead now i get:

![image](https://github.com/tteck/Proxmox/assets/7511738/5e7b0d4a-bf0c-4895-967a-22725c0c94de)

Any ideas?

BTW, @tteck great scripts! Any help very much appreciated.

**Reply by ViRouF** [reply]
did you solve this issue finally? Got the same

**Comment by CTMarsh** [comment]
Yeah. I had to reflash the sonoff in windows and then it just appeared when
connecting it back to the proxmox host.
